src/nerf_ffi.erl

```diff
@@ -1,6 +1,9 @@
 -module(nerf_ffi).
 
--export([ws_receive/2, ws_await_upgrade/2]).
+-export([ws_receive/2, ws_await_upgrade/2, ws_open/2]).
+
+% TODO: make the protocols a param that can be specified in gleam
```

Comment:
@lpil What would I accept here? And how would it work in gleam?

Reply:
We'll need to supply these options as an argument. The structure accepted can be seen here, but we could provide a minimal subset of that for now

https://github.com/ninenines/gun/blob/f9175998687678e227bdd49669e2d83f0648fa57/src/gun.erl#L136-L157

Perhaps something like this

```rust
pub type Options {
  Options(
    protocols: List(Protocol)
  )
}

pub type Protocol {
  Http
  Http2
  Raw 
  Socks
}
```

I'm also a little confused as to what is happening here, it looks like this is the default value in gun. I must be missing something, I am not familiar with the library. https://github.com/ninenines/gun/blob/f9175998687678e227bdd49669e2d83f0648fa57/src/gun.erl#L1048

Reply:
I provided separate methods to edit/change the protocols with the original methods doing as they did before. I believe tests should pass now and I can test in action soon.